### DISCUSSION TITLE
rspamd WebUi

### DISCUSSION THREAD

**Discussion by f0rc3r** [docker-mailserver/docker-mailserver#3560] [Q&A] [answered]
Hello there,

I got a Problem with the rspamd Web UI.
I made a custom-commands.conf like the example in the Documentation.
After I restart DMS, the custom-.commands.conf file is linked in the /etc/rspamd/override.d/ as custom-commands.conf with the same content. 

My problem is that when I go to the WebUi I still need to enter a Password. 

What did I miss ?

**Comment by polarathene** [comment]
> After I restart DMS

You probably restarted container `docker-compose restart` / `docker restart container_name`, this is not ok with DMS.

Use `docker-compose up --force-recreate`, or `docker-compose down && docker-compose up`.

---

If not resolving it, there is only one maintainer with rspamd experience and they're presently on vacation.

**Comment by f0rc3r** [comment]
Thanks for your Input.

I tried it, but I work almost completely with Portainer. I made a new Stack of DMS, but the Problem is still there.

**Reply by polarathene** [reply]
> but the Problem is still there.

I don't know how Portainer manages the containers. If it doesn't properly re-create DMS container from scratch and instead restarts the container (_which keeps internal state from last run present_), this can be problematic.

DMS startup scripts do a lot of config adjustments, many are done as modifications or additions with assumptions of a "clean slate" that we provide with the image. Nobody has tackled making this more resilient, only fixes here and there from time to time. If you are stuck with this situation you may experience various problems :(

---

For now I would suggest trying without Portainer involved, and see if you can at least get that working the way you want first. Then try transfer that to Portainer.

**Comment by f0rc3r** [comment]
Thx for the help, and sorry for the late reply.

I tried it, but it don't help. I still got this Issue.

**Comment by georglauterbach** [comment]
Posting the contents of the file would have been easy, right? I think this should work:

```txt
set-option-for-controller bind_socket "0.0.0.0:11334"
set-option-for-controller secure_ip "0.0.0.0/0"
```

**Comment by f0rc3r** [comment]
Hello again,

i copied the stuff from the Documentation in the file. It looks like this now:

![image](https://github.com/docker-mailserver/docker-mailserver/assets/109614681/37747c45-94ef-4839-b018-2336d73444c0)

and i still get asked about a Password when i try to get on the rSpamd Webinterface

**Comment by georglauterbach** [comment]
Please set `LOG_LEVEL=trace` and poste the log that contains `(Rspamd setup)`.

**Comment by f0rc3r** [comment]
Hello,

thats the Log with the rSpamd filter:

[  TRACE  ]  _setup_rspamd() registered
[  TRACE  ]  _start_daemon_rspamd_redis() registered
[  TRACE  ]  _start_daemon_rspamd() registered
[  DEBUG  ]  Enabling and configuring Rspamd
[  TRACE  ]  (Rspamd setup) ----------  Setup started  ----------
[  DEBUG  ]  (Rspamd setup) Found directory '/tmp/docker-mailserver/rspamd/override.d/' - linking it to '/etc/rspamd/override.d'
[  DEBUG  ]  (Rspamd setup) Internal Redis is enabled, adding configuration
[  DEBUG  ]  (Rspamd setup) Adjusting Postfix's configuration
[  DEBUG  ]  (Rspamd setup) Rspamd will not use ClamAV (which has not been enabled)
[  DEBUG  ]  (Rspamd setup) Disabling default modules
[  TRACE  ]  (Rspamd setup) Disabling module 'clickhouse'
[  TRACE  ]  (Rspamd setup) Disabling module 'elastic'
[  TRACE  ]  (Rspamd setup) Disabling module 'neural'
[  TRACE  ]  (Rspamd setup) Disabling module 'reputation'
[  TRACE  ]  (Rspamd setup) Disabling module 'spamassassin'
[  TRACE  ]  (Rspamd setup) Disabling module 'url_redirector'
[  TRACE  ]  (Rspamd setup) Disabling module 'metric_exporter'
[  DEBUG  ]  (Rspamd setup) Setting up intelligent learning of spam and ham
[  DEBUG  ]  (Rspamd setup) Enabling greylisting
[  DEBUG  ]  (Rspamd setup) Hfilter (group) module is enabled
[  TRACE  ]  (Rspamd setup) Not adjusting score for 'HFILTER_HOSTNAME_UNKNOWN' in Hfilter group module
[  TRACE  ]  (Rspamd setup) ----------  Setup finished  ----------
[  TRACE  ]  Destination /var/mail-state/lib-rspamd exists, linking /var/lib/rspamd to it
ENABLE_RSPAMD='1'
ENABLE_RSPAMD_REDIS='1'
RSPAMD_GREYLISTING='1'
RSPAMD_HFILTER='1'
RSPAMD_HFILTER_HOSTNAME_UNKNOWN_SCORE='6'
RSPAMD_LEARN='1'
[  DEBUG  ]  Starting rspamd-redis
[  DEBUG  ]  Starting rspamd

![image](https://github.com/docker-mailserver/docker-mailserver/assets/109614681/f4c247b7-d97c-4ea4-adf6-f224322753d0)

**Reply by georglauterbach** [reply]
Just as I thought: the file `custom-command.conf` is not parsed and applied. I can now see why: you use `/etc/rspamd/override.d/custom-commands.conf`, but the correct path is `/etc/rspamd/override.d/rspamd/custom-commands.conf`.

**Comment by f0rc3r** [comment]
Hello again,

i totally see now where are you pointing at. But these are some pictures where the volume and file are mounted and located. 

![Screenshot_20231027_182851_edit_392850224711857](https://github.com/docker-mailserver/docker-mailserver/assets/109614681/80a4999d-3c05-471b-9438-bca1085527f1)

![Screenshot_20231027_183005_edit_392928375737712](https://github.com/docker-mailserver/docker-mailserver/assets/109614681/1a3f095c-5d72-4a77-87e3-fd704b693287)

**Reply by georglauterbach** [reply]
It is not `.../config/rspamd/override.d/rspamd/custom-commands.conf`, just `.../config/rspamd/custom-commands.conf`.

**Comment by f0rc3r** [comment]
I put the file there because this is the part of the Documentation:

![image](https://github.com/docker-mailserver/docker-mailserver/assets/109614681/661ab937-bd9e-4640-95f8-bf6441f92979)


Now i put the custom-commands.conf here with the same results: 

![image](https://github.com/docker-mailserver/docker-mailserver/assets/109614681/c431f768-7a37-42f3-9e7f-faf3fb22e627)

and get these rspamd log:

[  TRACE  ]  _setup_rspamd() registered
[  TRACE  ]  _start_daemon_rspamd_redis() registered
[  TRACE  ]  _start_daemon_rspamd() registered
[  DEBUG  ]  Enabling and configuring Rspamd
[  TRACE  ]  (Rspamd setup) ----------  Setup started  ----------
[  DEBUG  ]  (Rspamd setup) Internal Redis is enabled, adding configuration
[  DEBUG  ]  (Rspamd setup) Adjusting Postfix's configuration
[  DEBUG  ]  (Rspamd setup) Rspamd will not use ClamAV (which has not been enabled)
[  DEBUG  ]  (Rspamd setup) Disabling default modules
[  TRACE  ]  (Rspamd setup) Disabling module 'clickhouse'
[  TRACE  ]  (Rspamd setup) Disabling module 'elastic'
[  TRACE  ]  (Rspamd setup) Disabling module 'neural'
[  TRACE  ]  (Rspamd setup) Disabling module 'reputation'
[  TRACE  ]  (Rspamd setup) Disabling module 'spamassassin'
[  TRACE  ]  (Rspamd setup) Disabling module 'url_redirector'
[  TRACE  ]  (Rspamd setup) Disabling module 'metric_exporter'
[  DEBUG  ]  (Rspamd setup) Setting up intelligent learning of spam and ham
[  DEBUG  ]  (Rspamd setup) Enabling greylisting
[  DEBUG  ]  (Rspamd setup) Hfilter (group) module is enabled
[  TRACE  ]  (Rspamd setup) Not adjusting score for 'HFILTER_HOSTNAME_UNKNOWN' in Hfilter group module
[  TRACE  ]  (Rspamd setup) ----------  Setup finished  ----------
[  TRACE  ]  Destination /var/mail-state/lib-rspamd exists, linking /var/lib/rspamd to it
ENABLE_RSPAMD='1'
ENABLE_RSPAMD_REDIS='1'
RSPAMD_GREYLISTING='1'
RSPAMD_HFILTER='1'
RSPAMD_HFILTER_HOSTNAME_UNKNOWN_SCORE='6'
RSPAMD_LEARN='1'
[  DEBUG  ]  Starting rspamd-redis
[  DEBUG  ]  Starting rspamd

**Reply by georglauterbach** [reply, marked as answer]
The docs are correct, please have a look again:

1. **When you want to use `custom-commands.conf`**: use `docker-data/dms/config/rspamd/custom-commands.conf` (see [this part of our documentation](https://docker-mailserver.github.io/docker-mailserver/edge/config/security/rspamd/#with-the-help-of-a-custom-file))
2. **When you want to manually supply files that override the Rspamd configuration**: use `docker-data/dms/config/rspamd/override.d/` (see [this part of our documentation](https://docker-mailserver.github.io/docker-mailserver/edge/config/security/rspamd/#manually))

**Note the difference in these paths**: for the first, it's just `.../config/rspamd/...`, for the second it is `...config/rspamd/override.d/...`!

---

PS: In [one of my earlier comments](https://github.com/orgs/docker-mailserver/discussions/3560#discussioncomment-7407653), I stated a wrong path - I am sorry for that. I corrected this comment.

PPS: Please format the logs better; here is a short link: <https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks>

**Comment by f0rc3r** [comment]
Great, it works now.
Thanks for your help. 

Sorry for the inconvenience I made. I appreciate your help very much.
